File: weekend-core/src/main/java/com/github/aly8246/core/query/AndBuilder.java
package com.github.aly8246.core.query;

import org.springframework.data.mongodb.core.query.Criteria;
import org.springframework.data.mongodb.core.query.Query;

import java.util.Arrays;

/**
 * @Author ：南有乔木
 * @Email ：1558146696@qq.com
 * @date ：Created in 2019/10/29 下午 05:40
 * @description：
 * @version: ：V
 */
public class AndBuilder implements CriteriaBuilder {
private Query query;

public AndBuilder(Query query) {
	this.query = query;
}

@Override
public Query buildQuery(String key, String value, CriteriaEnum criteriaEnum) {
	switch (criteriaEnum) {
		case GT://  >
			return query.addCriteria(Criteria.where(key).gt(value));
		case LT://  <
			return query.addCriteria(Criteria.where(key).lt(value));
		case EQ://  =
			return query.addCriteria(Criteria.where(key).is(value));
		case GE://  >=
			return query.addCriteria(Criteria.where(key).gte(value));
		case LE://  <=
			return query.addCriteria(Criteria.where(key).lte(value));
		case NQ:// !=
			return query.addCriteria(Criteria.where(key).ne(value));
		case IN:// in...
			return query.addCriteria(Criteria.where(key).in(Arrays.asList(value.split(","))));
	}
	return query;
}
}

File: weekend-core/src/main/java/com/github/aly8246/core/query/queryBuilder/AndBuilder.kt
package com.github.aly8246.core.query.queryBuilder

import com.github.aly8246.core.resolver.Condition
import com.github.aly8246.core.query.queryBuilder.basic.OperationSignCriteriaBuilder
import org.springframework.data.mongodb.core.query.Query


/**
 * @Author ：南有乔木
 * @Email ：1558146696@qq.com
 * @date ：Created in 2019/10/29 下午 05:40
 * @description：
 * @version: ：V
 */
class AndBuilder(private val query: Query) : QueryBuilder {

    override fun buildQuery(conditionList: List<Condition>): Query {
        for (conditions in conditionList) {
            val fieldName = conditions.fieldName
            val value = conditions.value
            val criteriaBuilder = OperationSignCriteriaBuilder()

            val criteria = criteriaBuilder.build(fieldName, value, conditions.sign)
            query.addCriteria(criteria)

        }
        return query
    }
}
